File: app/src/main/java/com/example/requestapp/ui/login/LoginActivity.java
package com.example.requestapp.ui.login;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.requestapp.R;
import com.example.requestapp.ui.fragment.TaskActivity;

public class LoginActivity extends AppCompatActivity implements LoginActivityContract.View {

    private TextView nick, password;
    private Button login;
    private ProgressBar progressBar;

    private LoginActivityContract.Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setPresenter();
        if (presenter.isLoggedIn()) {
            startActivity(new Intent(this, TaskActivity.class));
        }
        setContentView(R.layout.activity_login);
        initComponents();
        progressBar.setVisibility(View.INVISIBLE);
        login.setOnClickListener(onClickListener);
    }

    private void setPresenter() {
        presenter = new LoginActivityPresenter(this);
    }

    private void initComponents() {
        nick = findViewById(R.id.editTextNickName);
        password = findViewById(R.id.editTextPassword);
        login = findViewById(R.id.buttonLogin);
        progressBar = findViewById(R.id.progressBar);
    }

    @Override
    public void onLoginSuccessful(String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

    @Override
    public void onLoginFailure(String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

    @Override
    public void onProcessStart() {
        progressBar.setVisibility(View.VISIBLE);
    }

    @Override
    public void onProcessEnd() {
        progressBar.setVisibility(View.INVISIBLE);
    }

    private View.OnClickListener onClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            presenter.onLoginClicked(LoginActivity.this, nick.getText().toString(), password.getText().toString());
        }
    };
}

File: app/src/main/java/com/example/requestapp/ui/login/LoginActivity.kt
package com.example.requestapp.ui.login

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.example.requestapp.R
import com.example.requestapp.ui.fragment.TaskActivity

class LoginActivity : AppCompatActivity(), LoginActivityContract.View {

    private var nick: TextView? = null
    private var password: TextView? = null
    private var login: Button? = null
    private var progressBar: ProgressBar? = null

    private var presenter: LoginActivityContract.Presenter = LoginActivityPresenter(this, this)
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (presenter.isLoggedIn()) {
            finish()
            startActivity(Intent(this, TaskActivity::class.java))
        }
        setContentView(R.layout.activity_login)
        initComponents()
        progressBar!!.visibility = View.INVISIBLE
        login!!.setOnClickListener(onClickListener)
    }

    private fun initComponents() {
        nick= findViewById<EditText>(R.id.editTextNickName)
        password = findViewById<EditText>(R.id.editTextPassword)
        login = findViewById(R.id.buttonLogin)
        progressBar = findViewById<ProgressBar>(R.id.progressBar)
    }

    override fun onLoginSuccessful(message: String?) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
    }

    override fun onLoginFailure(message: String?) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
    }

    override fun onProcessStart() {
        progressBar!!.visibility = View.VISIBLE
    }

    override fun onProcessEnd() {
        progressBar!!.visibility = View.INVISIBLE
    }

    private val onClickListener = View.OnClickListener {
        presenter.onLoginClicked(this,
                nick!!.text.toString(),
                password!!.getText().toString())
    }

}